src/main/java/core/basesyntax/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport(Storage storage) {

```

Comment:
The method `getReport` should not directly take `Storage` as a parameter. Instead, it should work with a data structure that represents the current state of the fruits in the storage. This is to follow the encapsulation principle and to keep the `Storage` class hidden from other parts of the application.